switchboard/base.py

```diff
@@ -6,77 +6,91 @@
 :license: Apache License 2.0, see LICENSE for more details.
 """
 
-import time
-import logging
 import threading
 
-from .models import MongoModel
-from .signals import request_finished
-from .settings import settings
 
-log = logging.getLogger(__name__)
+class ModelDict(threading.local):
+    """
+    Dictionary-style access to model data.
+
+    If ``auto_create=True`` accessing modeldict[key] when key does not exist
+    will attempt to create it in the datastore.
 
+    Functions in two different ways, depending on the constructor:
+
+        # Given a model that has a attribute named ``foo`` where the value
+        # is "bar":
 
-NoValue = object()
+        mydict = ModelDict(Model, value='foo')
+        mydict['test']
```

Comment:
Where does "test" come from? It's not clear from your example, as it states that *a model that has a attribute named ``foo`` where the value is "bar"*.